### ISSUE TITLE
Not properly handled NullPointerException

### ISSUE THREAD

**Issue by steffsas** [CROSSINGTUD/CryptoAnalysis#316]
**Prerequisites**
Currently, I'm testing the CryptoAnalysis on non-crypto APIs. Therefore, I wrote a custom CrySL rule for java.util.stream.IntStream for testing purposes only. Custom CrySL rules are recognized.

**Bug**
Assume the following example:
![grafik](https://user-images.githubusercontent.com/32709406/99937573-5cb1c380-2d66-11eb-9e90-adf682bc0393.png)

And the following CrySL rule:
![grafik](https://user-images.githubusercontent.com/32709406/99937637-7b17bf00-2d66-11eb-95e3-211b65617130.png)

It seems there is an error with the type java.util.function.IntUnaryOperator **f** in the event iterate(i, **f**).

![grafik](https://user-images.githubusercontent.com/32709406/99937807-d34ec100-2d66-11eb-925f-cb3e19fc1995.png)

Due to the error, the static analysis is not done correctly, no errors are indicated at all. If the iterate() method does not appear in the ORDER section, the static analysis runs as expected.

**Comment by lukasngl** [comment]
Hi,
CogniCrypt does not check the Xtext validation, therefore liniking issues are not reported and the AST will be incomplete, i.e. have null pointers were they are not expected.
Your rule tries to link `iterate(i, f)` to the actual method of `IntStream`, but the implicit method signature you specified is `iterate(Integer, ...)` while it should be `interate(int, ...)` (see [Dokumentation](https://docs.oracle.com/javase/8/docs/api/java/util/stream/IntStream.html#iterate-int-java.util.function.IntUnaryOperator-) and failes silently.
Thus, the method of that event is a null pointer which is then dereferenced when it is converted to a String.

TL;DR: Rule is incorrect, i must be int not Integer.
